programs/programs/policyengine/policy_engine.py

```diff
@@ -105,35 +115,44 @@ def pe_input(screen: Screen, programs: List[PolicyEngineCalulator]):
             if hasattr(program, "pe_output_period") and Data in program.pe_outputs:
                 period = program.pe_output_period
 
-            if not Data.member:
-                data = Data(screen, members, relationship_map)
-                value = data.value()
-                unit = raw_input["household"][data.unit][data.sub_unit]
+            if issubclass(Data, Member):
```

Comment:
I'm having trouble understanding the logic here. Just my guess about `Data`, `pe_inputs` and `pe_outputs`.
`pe_inputs` and `pe_outputs` are something look like this`dependency.member.AgeDependency`, which hold the fields that policy engine looks for, like "age", "is_child" and "is_tax_unit_dependent", etc.
Don't know how to define `Data`, but should be something like this `dependency.member.AgeDependency` as well?
And when we check `if issubclass(Data, Member):`, the "member" in `dependency.member.AgeDependency` gives `True`, so we update everything under `raw_input`, but the "tax_units"? 
If "[data.unit]" is Member, then what does the "raw_input["household"][data.unit]" in  `unit = raw_input["household"][data.unit][member_id]` looks like?
Thanks!

Reply:
`Data` is a subclass of `PolicyEngineScreenInput`. `pe_inputs` and `pe_outputs` are lists of `Data`. The `issubclass` logic is used to check if the `Data` is a `Member` dependency because they are handled differently. The way PE handles multiple members is like:
```py
{
  "people": {
    "member_id_one": {...}, # member 1 data
    "member_id_two": {...}, # member 2 data
    ...
  }
}
```

Reply:
Thanks for taking the time answering my question! Giving me some idea of how our screener works with PE! Thanks!